server/src/com/thoughtworks/go/server/service/SecurityAuthConfigService.java

```diff
@@ -57,4 +62,34 @@ public void create(Username currentUser, SecurityAuthConfig securityAuthConfig,
         update(currentUser, securityAuthConfig, result, new SecurityAuthConfigCreateCommand(goConfigService, securityAuthConfig, authorizationExtension, currentUser, result));
     }
 
+
+    public void verifyConnection(SecurityAuthConfig securityAuthConfig, LocalizedOperationResult result) {
+        try {
+            ValidationResult validationResult = authorizationExtension.verifyConnection(securityAuthConfig.getPluginId(), securityAuthConfig.getConfigurationAsMap(true));
+            if (!validationResult.isSuccessful()) {
+                boolean isInvalidProfile = false;
+                for (ValidationError validationError : validationResult.getErrors()) {
+                    if (StringUtils.isBlank(validationError.getKey())) {
+                        result.stale(LocalizedMessage.string("CHECK_CONNECTION_FAILED", securityAuthConfig.getId(), validationError.getMessage()));
```

Comment:
`stale` is not the right status code here. For lack of a better code, can we use 422 instead?